src/routes/index.svelte

```diff
@@ -74,24 +77,24 @@
     showUcan({ target: { value: encodedUcan } })
   }
 
-  const showProof = (event: CustomEvent) => {
+  const showProof = async (event: CustomEvent) => {
     // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
     proofTree = event.detail.node
     encodedUcan = proofTree.token
-    decodedUcan = ucan.decode(encodedUcan)
+    decodedUcan = await ucan.decode(encodedUcan)
   }
 
   $: {
     if (decodedUcan) {
       ucan
-        .validate(decodedUcan)
-        .then(result => (validation = result))
+        .validate(encodedUcan)
+        .then(result => (validation = result.validation))
         .catch(() => (validation = null))
     }
   }
 
   $: isValid =
-    validation !== null &&
+    validation != null &&
```

Comment:
I generally like using `!= null`, because `undefined == null` (but not `undefined === null`).
Technically it shouldn't be possible that validation is undefined here, so it's totally correct to go for `!==`. It's mostly a habit of mine. What do you think?

Reply:
I think it's best to be strict, and make sure the value can only be one or the other. Or check for both. I like that it's explicit and you have to consider which cases you might have.